common/check_expired_dids

```diff
@@ -8,51 +8,41 @@
 # Authors:
 # - Vincent Garonne, <vincent.garonne@cern.ch>, 2013
 # - Thomas Beermann, <thomas.beermann@cern.ch>, 2019
-# - Eric Vaandering <ewv@fnal.gov>, 2020-2021
+# - Eric Vaandering <ewv@fnal.gov>, 2020-2022
+# - Maggie Voetberg <maggiev@fnal.gov>, 2024
 
 """
 Probe to check the backlog of expired dids.
 """
 from __future__ import print_function
-
 import sys
 import traceback
 from datetime import datetime
+from sqlalchemy import func
 
-from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
-from rucio.common.config import config_get
 from rucio.db.sqla import models
 from rucio.db.sqla.session import get_session
-from rucio.db.sqla.util import get_count
 
-from utils.common import probe_metrics
+from utils.common import PrometheusPusher
+
 
 # Exit statuses
 OK, WARNING, CRITICAL, UNKNOWN = 0, 1, 2, 3
 
-PROM_SERVERS = config_get('monitor', 'prometheus_servers', raise_exception=False, default='')
-if PROM_SERVERS != '':
-    PROM_SERVERS = PROM_SERVERS.split(',')
-
 if __name__ == "__main__":
     try:
-        registry = CollectorRegistry()
         session = get_session()
-        query = session.query(models.DataIdentifier.scope).filter(models.DataIdentifier.expired_at.isnot(None),
-                                                                  models.DataIdentifier.expired_at < datetime.utcnow())
-        result = get_count(query)
-        # Possible check against a threshold. If result > max_value then sys.exit(CRITICAL)
-        probe_metrics.gauge(name='undertaker.expired_dids').set(result)
-        Gauge('undertaker_expired_dids', '', registry=registry).set(result)
-
-        if len(PROM_SERVERS):
-            for server in PROM_SERVERS:
-                try:
-                    push_to_gateway(server.strip(), job='check_expired_dids', registry=registry)
-                except:
-                    continue
-
-        print(result)
+        with PrometheusPusher() as manager:
+            
+            query = (session.query(func.count(models.DataIdentifier.scope))
+                           .filter(models.DataIdentifier.expired_at.isnot(None),
+                                   models.DataIdentifier.expired_at < datetime.utcnow()))
+            result = query.scalar() or 0
+            # Possible check against a threshold. If result > max_value then sys.exit(CRITICAL)
+
+            manager.gauge('expired_dids.total',
```

Comment:
This does change the metric value. IIRC, ATLAS is ok with this in theory as we wanted to make things more sensical. But I'd like @dchristidis thoughts before we start using this for CMS.

Reply:
Yes, that should be generally acceptable. But please add a comprehensive list of such changes in the commit message.